aiohttp/helpers.py

```diff
@@ -367,6 +367,22 @@ def parse_mimetype(mimetype: str) -> MimeType:
     )
 
 
+@functools.lru_cache(maxsize=56)
+def parse_content_type(raw: str) -> Tuple[str, Dict[str, str]]:
+    """Parse Content-Type header.
+
+    Returns a tuple of the parsed content type and a dictionary of parameters.
+
+    Callers should be careful to make a copy of the returned dictionary if
+    they need to modify it, as the dictionary is shared between calls.
+    """
+    msg = HeaderParser().parsestr(f"Content-Type: {raw}")
+    content_type = msg.get_content_type()
+    params = msg.get_params(())
+    content_dict = dict(params[1:])  # First element is content type again
+    return content_type, content_dict
```

Comment:
I wonder if it's worth using MappingProxyType, to help ensure the copy() is always used below.

Reply:
I thought about doing that but than we can't use `dict.copy()` which is very well optimized in cpython and much faster than creating a fresh dict every time.  Since we are likely only going to use this internally I think there is little risk of mis-use.

Reply:
Is MappingProxyType.copy() not optimised then? I assumed it just passed the call to the underlying dict.

Reply:
```
>>> underlying_dict = {"x":"y"}
>>> from types import MappingProxyType
>>> mapping = MappingProxyType(underlying_dict)
>>> type(mapping.copy())
<class 'dict'>
```

I was incorrectly assuming we end up with a `MappingProxyType` on copy but it looks like a `dict`.

Testing performance now

Reply:
```
['copy mapping', 0.1910314169999765]
['copy dict', 0.17891329099984432]
```

Its a tiny bit slower.. but not enough to justify not doing it

Reply:
Switched to MappingProxyType.

Thanks for the quick look

Reply:
> I was incorrectly assuming we end up with a `MappingProxyType` on copy but it looks like a `dict`.

Given that it's a proxy, it would be a pointless method to clone the proxy object itself.